File: React_Native/React_Native/src/pages/menus/Sat.js
import React, {useEffect} from 'react';
import { useNavigation } from '@react-navigation/native';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';

const Sat = () => {
    const navigation = useNavigation();

    // useEffect(() => {
    //     console.log('ENTROU NA APLICIAÇÃO SAT!');

    //     return () => {
    //         console.log('SAIU DA APLCIAÇÃO SAT!');
    //     }
    // },[])

    const buttons = [
        {id: 1, text: 'ATIVAÇÃO SAT', onPress: () => navigation.navigate("AtivarSAT")},
        {id: 2, text: 'ASSOCIAR ASSINATURA', onPress: () => navigation.navigate("AssociarSAT")},
        {id: 3, text: 'TESTE SAT', onPress: () => navigation.navigate("TesteSAT")},
        {id: 4, text: 'CONFIGURAÇÕES DE REDE', onPress: () => navigation.navigate("ConfigSAT")},
        {id: 5, text: 'ALTERAR CÓDIGO DE ATIVAÇÃO', onPress: () => navigation.navigate("AlterarCodigoSAT")},
        {id: 6, text: 'OUTRAS FERRAMENTAS', onPress: () => navigation.navigate("FerramentasSAT")}
    ];

    return (
        <View style={styles.container}>
            <Text style={styles.textTitle}>GERSAT</Text>

            <View style={styles.containerButtons}>                

                {buttons.map(({text, onPress}, index) => (
                    <TouchableOpacity
                        key={index}
                        style={styles.button}
                        onPress={onPress}
                    >
                        <Text style={styles.textButton}>{text}</Text>
                    </TouchableOpacity>
                ))}
            </View>
        </View>
    )
};

const styles = StyleSheet.create({
    button: {
        width: 300,
        height: 40,
        backgroundColor: '#c7c7c7',
        justifyContent: 'center',
        alignItems: 'center',
        marginBottom: 10,
        borderRadius: 5
    },

    container: {
        flex: 1,
        justifyContent: 'flex-start',
        alignItems: 'center'
    },
    containerButtons: {
        width: '35%',
        alignItems: 'center',
        justifyContent: 'center',
    },

    textButton: {
        fontSize: 14,
        fontWeight: 'bold'
    },
    textTitle: {
        fontSize: 40,
        fontWeight: 'bold',
        color: '#4a524c',
        marginVertical: 30
    }
});

export default Sat;

File: Flutter/Flutter/lib/pages/menus/sat.dart
import 'package:flutter/material.dart';
import 'package:flutter_gBot/pages/sat_pages/ferramentasSat.dart';
import 'package:flutter_gBot/widgets/widgetsgertec.dart';

import '../../pages/sat_pages/associarSat.dart';
import '../../pages/sat_pages/ativarSat.dart';
import '../../pages/sat_pages/configRede.dart';
import '../../pages/sat_pages/ferramentasSat.dart';
import '../../pages/sat_pages/testeSat.dart';
import '../../pages/sat_pages/alterarCodigo.dart';

class PageSat extends StatefulWidget {
  @override
  _PageSatState createState() => _PageSatState();
}

// Tela de Menu Principal do Sat com suas Funções principais
class _PageSatState extends State<PageSat> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SingleChildScrollView(
        child: Container(
          color: Colors.white,
          padding: EdgeInsets.only(top: 30),
          height: MediaQuery.of(context).size.height,
          width: MediaQuery.of(context).size.width,
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.center,
            mainAxisSize: MainAxisSize.min,
            children: <Widget>[
              Text(
                "GERSAT",
                style: TextStyle(fontWeight: FontWeight.bold, fontSize: 45),
              ),
              WidgetsGertec.buttonStandard(
                "ATIVAÇÃO SAT",
                callback: () {
                  Navigator.push(
                    context,
                    MaterialPageRoute(builder: (context) => PageAtivarSat()),
                  );
                },
              ),
              WidgetsGertec.buttonStandard(
                "ASSOCIAR ASSINATURA",
                callback: () {
                  Navigator.push(
                    context,
                    MaterialPageRoute(builder: (context) => PageAssociarSat()),
                  );
                },
              ),
              WidgetsGertec.buttonStandard(
                "TESTE SAT",
                callback: () {
                  Navigator.push(
                    context,
                    MaterialPageRoute(builder: (context) => PageTesteSat()),
                  );
                },
              ),
              WidgetsGertec.buttonStandard(
                "CONFIGURAÇÕES DE REDE",
                callback: () {
                  Navigator.push(
                    context,
                    MaterialPageRoute(builder: (context) => PageConfigSat()),
                  );
                },
              ),
              WidgetsGertec.buttonStandard(
                "ALTERAR CÓDIGO DE ATIVAÇÃO",
                callback: () {
                  Navigator.push(
                    context,
                    MaterialPageRoute(builder: (context) => PageCodigoSat()),
                  );
                },
              ),
              WidgetsGertec.buttonStandard(
                "OUTRAS FERRAMENTAS",
                callback: () {
                  Navigator.push(
                    context,
                    MaterialPageRoute(builder: (context) => PageFerramentaSat()),
                  );
                },
              )
            ],
          ),
        ),
      ),
    );
  }
}
